crates/binstalk/src/ops/resolve.rs

```diff
@@ -423,6 +384,101 @@ fn collect_bin_files(
     Ok(bin_files)
 }
 
+struct PackageInfo {
+    meta: PkgMeta,
+    binaries: Vec<Bin>,
+    name: CompactString,
+    version_str: CompactString,
+    version: Version,
+    repo: Option<String>,
+    overrides: BTreeMap<String, PkgOverride>,
+}
+
+struct Bin {
+    name: String,
+    required_features: Vec<String>,
+}
+
+impl PackageInfo {
+    /// Return `None` if already up-to-date.
+    async fn resolve(
+        opts: &Options,
+        name: CompactString,
+        curr_version: Option<Version>,
+        version_req: VersionReq,
+        client: Client,
+        crates_io_api_client: crates_io_api::AsyncClient,
+    ) -> Result<Option<Self>, BinstallError> {
+        // Fetch crate via crates.io, git, or use a local manifest path
+        // TODO: work out which of these to do based on `opts.name`
+        // TODO: support git-based fetches (whole repo name rather than just crate name)
```

Comment:
let's remove these todos. we've got the local manifest support via an option, and git repo support is covered in #3.

Reply:
Done